### PR TITLE
Results- hetionet reconstruction

### PR DESCRIPTION
**Pull request by danich1**
This section talks about how many edges can be recalled/added to Hetionet.

### CODE REVIEW

File: content/04.results.md [hunk 1]
```diff
@@ -10,4 +10,18 @@ place the grid of aurocs here for discriminator model
 place the results of random label function experiment
 
 ## Reconstructing Hetionet
-place figure of number of new edges that can be added to hetionet as well as edges we can reconstruct using this method
+After the discriminiative model has been calibrated, we estimated the number of edges that can be added/recalled from Hetionet v1.
+Figure {@fig:hetionet_reconstruction}, depicts the number of edges based on precision level.
+For Disease associates Gene (DaG) we can add 74 new edges while matintaining 70% precision.
```
Comment (dhimmel) [inline content/04.results.md]: We usually do "Disease-associates-Gene"

File: content/04.results.md [hunk 2]
```diff
@@ -10,4 +10,18 @@ place the grid of aurocs here for discriminator model
...
+Likewise with Compound treats Disease (CtD) we can add 10 new edges while maintaining 70% precision.
+As for Compound binds Gene (CbG) and Gene interacts Gene (GiG) the models fail to reach precision higher than 0.5 and 0.6 respectively.
+Despite the low performance, we are still able to add new edges (7 for CbG at 47% precision and 9 for GiG at 56% precision) to Hetionet v1.
```
Comment (dhimmel) [inline content/04.results.md]: How is precision determined here? Since you've done the probability calibration, could you just use the predicted probability as the precision of that point? Also note that 70% precision could actually be quite high if Hetionet is incomplete. In these cases, all your top predictions could be true, but yet you may still get lower precision since the truth is not completely in Hetionet.

Comment (danich1) [reply content/04.results.md]: Precision is calculated using already established edges as positive while never before seen edges are treated as negative. You raise a good point as there are some edges that might be true even though they haven't been seen in Hetionet.

Comment (cgreene) [reply content/04.results.md]: Is this performance at recovering het.io on the test set?

**Comment by cgreene** [comment]
Screen Shot 2019-07-24 at 11 08 36 AM

Can't see the "literature model." Also, why are you calling it a recently published literature model? What does literature model mean in this context?

**Review by cgreene — COMMENTED** [review]
I'm still getting the same error on firefox: the image contains errors.

**Comment by danich1** [comment]
That is weird. The image shows up on my computer and on my laptop (both firefox and chrome). Will look into it. But here is a link to the [image itself](https://raw.githubusercontent.com/greenelab/snorkeling/master/figures/literature_models/model_comparison.png).

File: content/04.results.md (hunk 2 above)
Comment (danich1) [reply content/04.results.md]: Performance here is using all edges in het.io as the evaluation set. After each sentence has been scored, I combined them together based on mention pair and take the max score for each pair.

Comment (cgreene) [reply content/04.results.md]: Since het.io is used for distant supervision, is there an issue with this + evaluation?

Comment (danich1) [reply content/04.results.md]: Hmmm..  Based on past experiences, my intuition says there is an overfitting problem or at least there is a confounding issue with the evaluation. If I'm on the right path, I should have "held out" edges from distant supervision label functions and see if my method can predict the "held out" edges downstream.

Comment (cgreene) [reply content/04.results.md]: What if you evaluate this on only edges from the test set predictions?

File: content/04.results.md [hunk 3]
```diff
@@ -9,5 +9,36 @@ place the grid of aurocs here for discriminator model
 ## Random Noise Generative Model
 place the results of random label function experiment
 
-## Reconstructing Hetionet
-place figure of number of new edges that can be added to hetionet as well as edges we can reconstruct using this method
+
+### Reconstructing Hetionet
+Once the discriminiative model has been calibrated, we compared our model to a baseline model [@tag:cocoscore] (Figure {@fig:cocoscore_comparison}) and  estimated the number of edges that can be added/recalled from Hetionet v1 (Figure Figure {@fig:hetionet_reconstruction}).
```
Comment (cgreene) [inline content/04.results.md]: Does it matter that you use the calibrated model here? I don't think you're using the probabilities if I understand correctly.

Also, since the calibration may or may not have worked (kinda unclear from those plots if it's really better calibrated - why use the calibrated edges here? I could understand why you would if it worked - because then you could directly used the probabilities emitted from the model. I'm just not sure why it's used here if the calibration didn't really work.

Comment (cgreene) [inline content/04.results.md]:
```suggestion
Once the discriminiative model has been calibrated, we compared our model to a baseline model [@tag:cocoscore] (Figure {@fig:cocoscore_comparison}) and  estimated the number of edges that can be added/recalled from Hetionet v1 (Figure {@fig:hetionet_reconstruction}).
```

Comment (cgreene) [reply content/04.results.md]: Also, I'm not sure why the comparison and adding edges are stuck together as a combined section. Is there something that links them?

File: content/04.results.md [hunk 4]
```diff
@@ -9,5 +9,36 @@ place the grid of aurocs here for discriminator model
...
+Regarding model comparison we evaluated edge prediction performance using all edges assigned to our test set.
+Scores for both models are reported in the form of area under the receiver operative curve (AUROC) and area under the precision recall curve (AUPR).
```
Comment (cgreene) [inline content/04.results.md]:
```suggestion
We report performance for both models with AUROC and AUPR.
```
Only define acronyms the first time.

File: content/04.results.md [hunk 5]
```diff
@@ -9,5 +9,36 @@ place the grid of aurocs here for discriminator model
...
+The baseline model outperforms our model for majority of edge types.
+Our model achieves an AUROC of 78% and an AUPR of 28% for Disease-associates-Gene (DaG), while the baseline model achieves a score of 82% and 45% resepctively.
```
Comment (cgreene) [inline content/04.results.md]:
```suggestion
Our model achieves an AUROC of 78% and an AUPR of 28% for DaG, while the previously published model achieves a score of 82% and 45% respectively.
```

File: content/04.results.md [hunk 6]
```diff
@@ -9,5 +9,36 @@ place the grid of aurocs here for discriminator model
 ## Random Noise Generative Model
 place the results of random label function experiment
 
-## Reconstructing Hetionet
-place figure of number of new edges that can be added to hetionet as well as edges we can reconstruct using this method
+
+### Reconstructing Hetionet
+Once the discriminiative model has been calibrated, we compared our model to a baseline model [@tag:cocoscore] (Figure {@fig:cocoscore_comparison}) and  estimated the number of edges that can be added/recalled from Hetionet v1 (Figure Figure {@fig:hetionet_reconstruction}).
+Regarding model comparison we evaluated edge prediction performance using all edges assigned to our test set.
```
Comment (cgreene) [inline content/04.results.md]:
```suggestion
We evaluated edge prediction performance using all edges assigned to our test set.
```

File: content/04.results.md [hunk 7]
```diff
@@ -9,5 +9,36 @@ place the grid of aurocs here for discriminator model
...
+Scores for both models are reported in the form of area under the receiver operative curve (AUROC) and area under the precision recall curve (AUPR).
+The baseline model outperforms our model for majority of edge types.
+Our model achieves an AUROC of 78% and an AUPR of 28% for Disease-associates-Gene (DaG), while the baseline model achieves a score of 82% and 45% resepctively.
+For Gene-interacts-Gene the baseline model achieved an AUROC of 73% and an AUPR of 11%, while our model achieved an AUROC of 72% and an AUPR of 7%.
```
Comment (cgreene) [inline content/04.results.md]:
```suggestion
For Gene-interacts-Gene the baseline model achieved an AUROC of 73% and an AUPR of 11%, while our model achieved an AUROC of 72% and an AUPR of 7%.
```

File: content/04.results.md [hunk 8]
```diff
@@ -9,5 +9,36 @@ place the grid of aurocs here for discriminator model
...
+Our model out performed the literature model for Compound-treats-Disease (CtD) by obtaining an AUROC of 94% and an AUPR of 40%.
```
Comment (cgreene) [inline content/04.results.md]: All of these things are apparent from the figure. You don't need to repeat things that are exactly in the figure in a text form. Can you summarize the main interpretation here? See what I did in #15 for discriminative model performance.

**Review by cgreene — CHANGES_REQUESTED** [review]
This needs some work to summarize the take-away message from the figures instead of just reporting the numbers that are already represented there. There may also be some structural challenges, or else there is some link between the two things that are joined here that I don't quite understand.

**Comment by dhimmel** [comment]
Is the diff clean (only showing changes that this PR creates)? If not, I can help @danich1 fix before I review.

**Comment by danich1** [comment]
Turns out I've been working on everything from my laptop. My desktop version was outdated. Fixed now and updating the desktop version. Ready for review @dhimmel.

File: content/04.results.md [hunk 9]
```diff
@@ -47,21 +47,21 @@ The last most column consists of pooling every relationship specific label funct
...
 However, in general the performance changes were smaller than those observed with mismatched label types.
 
 ![
 A grid of area under the precision recall curve (AUPR) for each edge type.
 Each plot consists of adding a single label function on top of the baseline model.
-This label function emits a positive (top row) or negative (bottom row) label at specified frequencies.
+This label function emits a positive (shown in blue) or negative (shown in orange) label at specified frequencies.
 The error bars represent 95% confidence intervals for AUPR (y-axis) at emission frequency.
-](https://raw.githubusercontent.com/danich1/snorkeling/f8962788e462b783be05a6dec5eec7fe0f0259e7/figures/gen_model_error_analysis/transfer_test_set_auprc.png){#fig:random_label_function_auprc}
+](https://raw.githubusercontent.com/greenelab/snorkeling/master/figures/gen_model_error_analysis/transfer_test_set_auprc.png){#fig:random_label_function_auprc}
```
Comment (dhimmel) [inline content/04.results.md:64]: Usually I suggest versioned links because when the image changes, oftentimes your manuscript should too.

File: content/04.results.md [hunk 10]
```diff
@@ -112,5 +112,32 @@ A perfectly calibrated model would follow the dashed diagonal line.
...
+Across edge types our model shows comparable performance against the baseline in terms of AUROC.
+Regarding AUPR, our model shows hindered performance against the baseline.
+The exception for both cases is CtD where our model performs better than the baseline.
+
+![
+Comparion between our model and CoCoScore model [@tag:cocoscore].
+We report both model's performance in terms of area under the receiver operating curve (AUROC) and area under then precision recall curve (AUPR).
+Our model achieves comparable performance against CoCoScore in terms of AUROC.
+As for AUPR CoCoScore consistently outperforms our model except for compound treats disease (CtD). 
+](https://raw.githubusercontent.com/greenelab/snorkeling/master/figures/literature_models/model_comparison.png){#fig:cocoscore_comparison}
```
Comment (dhimmel) [inline content/04.results.md]: For the figure legend is there a name for "our model"? For example, if someone else were to show this figure in their presentation, "Our Model" would make less sense.

The AUROC bar chart should have a y-axis from 0.5 to 1.

File: content/04.results.md [hunk 11]
```diff
@@ -112,5 +112,32 @@ A perfectly calibrated model would follow the dashed diagonal line.
...
+We evaluated how many edges we can recall and add to Hetionet v1 (Figure {@fig:hetionet_reconstruction}).
+In our evaluation we used edges assigned to our test set.
+Overall, we can recall a small amount of edges at high precision thresholds.
+A key example is CbG and GiG where we recalled only one exisiting edge at 100% precision.
+Despite the low recall, we are still able to add novel edges to DaG and CtD while retaining modest precision.
+
+![
+A scatter plot showing the number of edges (log scale) we can add or recall at specified precision levels. 
+The blue depicts edges exisiting in hetionet and the orange depicts how many novel edges can be added.
+](https://raw.githubusercontent.com/greenelab/snorkeling/master/figures/edge_prediction_experiment/edges_added.png){#fig:hetionet_reconstruction}
```
Comment (dhimmel) [inline content/04.results.md]: Have you tried adding lines between the points in this plot?

File: content/04.results.md [hunk 12]
```diff
@@ -112,5 +112,32 @@ A perfectly calibrated model would follow the dashed diagonal line.
 The blue line represents the predictions before calibration and the orange line shows predictions after calibration. 
 ](https://raw.githubusercontent.com/greenelab/snorkeling/master/figures/model_calibration_experiment/model_calibration.png){#fig:discriminative_model_calibration}
 
-## Reconstructing Hetionet
-place figure of number of new edges that can be added to hetionet as well as edges we can reconstruct using this method
+
+### Baseline Comparison
+Once our discriminator model is calibrated, we compared our model's ability to predict edges in our test set to a previosuly published baseline model [@tag:cocoscore].
```
Comment (dhimmel) [inline content/04.results.md]: The discriminator provides predictions for sentences... how do you convert from sentence predictions to edge predictions?

File: content/04.results.md [hunk 13]
```diff
@@ -112,5 +112,32 @@ A perfectly calibrated model would follow the dashed diagonal line.
...
+Overall, we can recall a small amount of edges at high precision thresholds.
+A key example is CbG and GiG where we recalled only one exisiting edge at 100% precision.
+Despite the low recall, we are still able to add novel edges to DaG and CtD while retaining modest precision.
+
+![
+A scatter plot showing the number of edges (log scale) we can add or recall at specified precision levels. 
+The blue depicts edges exisiting in hetionet and the orange depicts how many novel edges can be added.
+](https://raw.githubusercontent.com/greenelab/snorkeling/master/figures/edge_prediction_experiment/edges_added.png){#fig:hetionet_reconstruction}
+
+
```
Comment (dhimmel) [inline content/04.results.md:144]: To help people understand the output and its promise, would it be a good idea to include a table of top predictions?

You could image the following columns:

- edge type
- source node
- target node
- probability (generator)
- probability (discriminator)
- number of sentences
- example sentence

This table will vary depending on what exactly has been done relating to my above question:

> The discriminator provides predictions for sentences... how do you convert from sentence predictions to edge predictions?

File: content/90.back-matter.md [hunk 14]
```diff
@@ -6,7 +6,54 @@
 <style> span.gene_color { color:#02b3e4 } span.disease_color { color:#875442 } span.compound_color { color:#e91e63 } </style> 
 ## Supplemental Figures {.page_break_before}
 
-### Confidence Scores for Each Edge Type 
+### Top Edge Prediction Tables
```
Comment (dhimmel) [inline content/90.back-matter.md:9]: Table looks great. Excited to see how it renders with the coloring... could be very slick. We may consider adding a smaller version to the main text, like with 5-10 rows, but we can decide later

**Review by dhimmel — APPROVED** [review]
